core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -473,6 +481,28 @@ public void start(Boolean isLeader) throws IOException {
           HeartbeatContext.MASTER_LOST_WORKER_DETECTION, new LostWorkerDetectionHeartbeatExecutor(),
           (int) Configuration.getMs(PropertyKey.MASTER_LOST_WORKER_DETECTION_INTERVAL),
           Configuration.global(), mMasterContext.getUserState()));
+
```

Comment:
Thanks, what do you think about instead of making this an timed executor, make it a sleeping thread that is awoken by the caller that observes mBlockContainerIdGenerator.getNextContainerId()
               == (mJournaledNextContainerId - mContainerIdReservationSize / 2)?

Reply:
It has been changed from scheduled tasks to submitting tasks to single thread pool. The reason why a single thread pool is used instead of a single thread is because the implementation of the thread needs to consider interruption and awake(wait and signal?), it requires more complex code to achieve the same function.